api/src/main/java/org/openmrs/module/fgh/mpi/FhirUtils.java

```diff
@@ -670,6 +639,96 @@ private synchronized static void initializeCachesIfNecessary() {
 				}
 			}
 		}
+		
+		if (santeMessageHeaderEventUri == null) {
+			santeMessageHeaderEventUri = MpiUtils.getGlobalPropertyValue(GP_SANTE_MESSAGE_HEADER_EVENT_URI);
+		}
+		
+		if (santeMessageHeaderFocusReference == null) {
+			santeMessageHeaderFocusReference = MpiUtils.getGlobalPropertyValue(GP_SANTE_MESSAGE_HEADER_FOCUS_REFERENCE);
+		}
+	}
+	
+	/**
+	 * Generates a fhir map for Message Header needed by santeMPI when submit a bundle
+	 * 
+	 * @return a map containing the the message header objects
+	 */
+	public static Map<String, Object> generateMessageHeader() {
```

Comment:
FYI, when I tested SanteMPI, I didn't see where this is required, I was able to submit a plain patient with no header, do you really need this?

Reply:
Yes, we need it when we submit a bundle.